src/common/framework/query/compare.ts

```diff
@@ -44,13 +40,10 @@ export function compareQueries(a: TestQuery, b: TestQuery): Ordering {
 // "IsBig" means the query is big relative to the level, e.g. for test-level:
 //   anything >= suite:a,* is big
 //   anything <= suite:a:* is small
-function compareOneLevel(
-  ordering: Ordering,
-  aIsBig: boolean,
-  bIsBig: boolean
-): Ordering | undefined {
+function compareOneLevel(ordering: Ordering, aIsBig: boolean, bIsBig: boolean): Ordering {
+  assert(ordering !== Ordering.Equal || aIsBig || bIsBig);
   if (!aIsBig && !bIsBig) {
```

Comment:
I think the logic here could be made a bit simpler, see previous comment on this method